### PR TITLE
Modes enabling

### PR DESCRIPTION
**Pull request by eSqadron**
Possibility to enable and disable specific modes added to KConfig.

### CODE REVIEW

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+config POS_CONTROL_ENABLE
+	bool "Enable functions responsible for position control, like mode position, \
+	     pos set, etc.. Continuus calculation of position and pos read is always \
+	     enabled."
+	default y
+
 config KP_NUMERATOR_FOR_SPEED_IN_POS
 	int "proportional part numerator for setting speed during position control"
 	default 4
+	# depends on POS_CONTROL_ENABLE
```
Comment (ghost) [inline BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl]: why is that commented out?

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl [hunk 2]
```diff
@@ -25,14 +35,16 @@ config POS_CONTROL_PRECISION_MODIFIER
...
 	int "Histeresis percentage of position control. For no histeresis, set to 100."
 	default 100
+	# depends on POS_CONTROL_ENABLE
 
 config POS_SOFT_START
 	bool "Speed during position control will incrementally start from zero. If false,\
 	will immediately start from given speed."
 	default y
-
+	# depends on POS_CONTROL_ENABLE
```
Comment (ghost) [inline BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl]:
```suggestion
	# depends on POS_CONTROL_ENABLE

```

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl [hunk 3]
```diff
@@ -1,20 +1,30 @@
 menu "Position Control Configuration"
 
+config POS_CONTROL_ENABLE
```
Comment (ghost) [inline BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl:3]: if there are multiple configs in the same config, you can use:
```
if POS_CONTROL_ENABLE

config AAA
    bool "AAA description"
    default y


config BBB
    int "BBB description"
    default 123


endif
```

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.speed_cntrl [hunk 4]
```diff
@@ -1,11 +1,19 @@
 menu "Speed Control Configuration"
 
+config SPEED_CONTROL_ENABLE
```
Comment (ghost) [inline BrushedMotorControllerFirmware/Kconfigs/Kconfig.speed_cntrl:3]: same as above, please use `if` statement for multiple configs that depend on the same config

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.speed_cntrl [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
+	bool "Enable functions responsible for speed control, like mode speed, \
+	     speed set, etc.. Continuus calculation of speed and speed read \
+	     is always enabled."
```
Comment (ghost) [inline BrushedMotorControllerFirmware/Kconfigs/Kconfig.speed_cntrl]:
```suggestion
	bool "Enable functions responsible for speed control, like mode speed, \
	     speed set, etc."
```

imo there is no point to explain that continuous speed calculation is always enabled. dependant configs are descriptive well enough, and speed read will be visible from shell

File: BrushedMotorControllerFirmware/include/driver.c [hunk 6]
```diff
@@ -38,7 +38,7 @@ static int set_direction_raw(enum MotorDirection direction, enum ChannelNumber c
 #pragma endregion InternalFunctions
 
 /// CONTROL MODE - whether speed or position is controlled
-static enum ControlModes control_mode = SPEED;
+static enum ControlModes control_mode = (enum ControlModes)0;//Default to first defined control mode
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/driver.c]: static  uninitialised variables are always defaulted with `0` value
```suggestion
static enum ControlModes control_mode;
```

File: BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c [hunk 7]
```diff
@@ -140,17 +146,45 @@ static int cmd_actual_direction(const struct shell *shell, size_t argc, char *ar
 }
 
 SHELL_CMD_ARG_REGISTER(speed, NULL,
-	"speed in milli RPM (one thousands of RPM)\nspeed to get speed\n speed <val> to set speed",
-	cmd_speed, 1, 1);
+#if defined(CONFIG_SPEED_CONTROL_ENABLE)
+			"Speed in milli RPM (one thousands of RPM).\n"
+			"speed to get current motor speed.\n speed <val> to set new target speed.",
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c]:
```suggestion
			"speed to get current motor speed.\n"
			"speed <val> to set new target speed.",
```

to make it more readable from src code

File: BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c [hunk 8]
```diff
@@ -140,17 +146,45 @@ static int cmd_actual_direction(const struct shell *shell, size_t argc, char *ar
...
 SHELL_SUBCMD_SET_END
 );
+#endif
 
-SHELL_CMD_ARG_REGISTER(pos, &position_tree, "get/set position in deca-degree", cmd_position, 1, 1);
+SHELL_CMD_ARG_REGISTER(pos,
+#if defined(CONFIG_POS_CONTROL_ENABLE)
+			&position_tree,
+			"Motor position in deca-degrees (one hundreds of degree).\n"
+			"pos to get current position.\n pos <val> to set new target position.",
```
Comment (ghost) [inline BrushedMotorControllerFirmware/include/uart_shell/shell_direct_motor_control.c]:
```suggestion
			"pos to get current position.\n"
			"pos <val> to set new target position.",
```

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl (hunk 1 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl]: leftover from debugging; corrected, including all implications!

File: BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl (hunk 3 above)
Comment (eSqadron) [reply BrushedMotorControllerFirmware/Kconfigs/Kconfig.position_cntrl:3]: corrected

**Review by ghost — APPROVED** [review]

**Pull request merged**